### ISSUE TITLE
[testing] Allow for accessibility testing

### ISSUE THREAD

**Issue by jarrodek** [salesforce/lwc#1814]
**Is your feature request related to a problem? Please describe.**
Accessibility tests should be an integral part of unit testing. Currently there is no in-framework way of performing a11y testing. At least non that I am aware of.

**Describe the solution you'd like**
Some time ago I have built a module to perform a11y tests for [@open-wc](https://github.com/open-wc/open-wc/tree/master/packages/chai-a11y-axe). It extends the `Assertion` interface of Chai framework. This module is integrated with open-wc's test framework so there's no effort from the developer side to perform a11y testing. I would like to see similar solution in LWC so I don't have to do it manually for each project I am working on.

**Describe alternatives you've considered**
This can be done manually but it doesn't scale. I could technically have own package but it wouldn't be integrated with the the testing framework and other developers would also benefit from it.

**Comment by caridy** [comment]
I don't fully understand how is this different from any other WC framework. Lets first state that WC has many issues with accessibility, and LWC does do much to try to help. We do have some basic integration testings for that using karma (you can look at them in the karma folder). Jest on the other hand is a pain when it comes to focus tests, improving, but still challenging.

If I understand correctly, what you're looking for is an abstraction to write accessibility tests easier for LWC components, but I think something for WC is sufficient here. You can build such abstraction outside of LWC umbrella, and if it picks up, we can make it part of our recommendation for such tests. cc @ekashida which is our accessibility/focus expert.

**Comment by jarrodek** [comment]
I am not asking for a11y support in components. It's in title. I am asking for a11y test support in testing, similarly to the one I designed for open-wc, so not only I have a package and use it but all users can benefit of having integrated tests available without an effort.

**Comment by caridy** [comment]
closing this after speaking to @jarrodek offline. a tool can be created on top of web components or lwc. If there is anything that needs to be done at the LWC level, a new issue can be opened.
